File: transaction&concurrency/concurrency/src/main/java/com/example/concurrency/domain/order/service/OrderService.java
package com.example.concurrency.domain.order.service;

import com.example.concurrency.domain.item.entity.Item;
import com.example.concurrency.domain.order.dto.OrderDto;
import com.example.concurrency.domain.order.entity.Order;
import com.example.concurrency.domain.order.repo.OrderRepository;
import com.example.concurrency.domain.user.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class OrderService {
    private final OrderRepository orderRepository;

    @Transactional
    public Order createOrder(User user, Item item, OrderDto orderDto) {
        Order order = Order.builder().count(orderDto.getCount())
                .totalPrice(orderDto.getTotalPrice())
                .registerDate(orderDto.getRegisterDate())
                .build();

        order.setOrderWithUser(user);
        order.setItem(item);

        return orderRepository.save(order);
    }

    public void deleteAllItems(){
        orderRepository.deleteAllItems();
    }

    public long getOrderCount(){
        return orderRepository.count();
    }
}

File: Hexagonal/hexagonal/src/main/kotlin/com/example/hexagonal/core/service/order/OrderService.kt
package com.example.hexagonal.core.service.order

import com.example.hexagonal.core.domain.order.Order
import com.example.hexagonal.core.domain.order.OrderItem
import com.example.hexagonal.ports.`in`.usecase.order.OrderUseCase
import com.example.hexagonal.ports.out.order.OrderRepositoryPort
import org.springframework.stereotype.Service

@Service
class OrderService (
    private val orderRepository: OrderRepositoryPort
) : OrderUseCase
{
    override fun createOrder(order: Order): Order {
        return orderRepository.save(order)
    }

    override fun getAllOrders(): List<Order> {
        return orderRepository.findAll()
    }

    override fun getOrderById(id: Long): Order? {
        return orderRepository.findById(id)
    }

}
